text/0000-more-env-info.md

```diff
@@ -0,0 +1,58 @@
+- Feature Name: more-env-info
+- Start Date: 2017-08-09
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Introduce more potentially useful environment-related functions to `std::env`.
+
+# Motivation
+[motivation]: #motivation
+
+Programs sometimes need to know about their working environment to do their job properly. For example, linebreak convention differs for Windows and *NIX. Such discrepancy can lead to problems easily, especially when a program needs to communicate with aged third-party libraries.
+
+As a system programming language, it would be good for Rust to know the system it works on.
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC would like to introduce the following functions to `std::env`:
+
+```rust
+/// Platform architecture.
+/// e.g. `arm`, `x86_64` and `i686`.
+pub fn arch() -> String;
+/// Conventional linebreak of current platform.
+pub fn linebreak() -> String;
+/// Current operating system.
+/// e.g. `windows`, `linux` and `darwin`.
+pub fn os() -> String;
+/// Word size in bits the program has been compiled into.
+/// Commonly 32 or 64.
+pub fn word_size() -> u32;
```

Comment:
Is it, though? `usize` is defined as a 'pointer-sized unsigned integer type', which connects it to the bit width of the memory address space, not of a general-purpose register (which need not be the same). Given this definition, under the x32 ABI for x86-64 `usize` ought to be 32-bit, even though general-purpose registers are 64-bit. I also wonder how `usize` should be defined on 32-bit x86 with segmented pointers...

I think there should be a separate integer type with the bit width of a general-purpose/arithmetic register.